### PR TITLE
[SwipeRefresh] Make indicator more flexible

### PR DESCRIPTION
**Pull request by chrisbanes**
There are a few improvements here:

- Added more parameters to `SwipeRefresh()` to tweak the positioning and padding of the indicator. This allows apps which use `contentPadding` on `LazyRow` etc, to get the indicator to match.
- Move all positioning of the indicator to `SwipeRefreshIndicator()` itself. This allows devs to provide any type of indicator they want. The update custom indiciator sample shows a fade-in gradient instead demo.
- Remove the 'refresh trigger' value from `SwipeRefreshState`. It's never really been part of the state, and was only there for convenience.

### CODE REVIEW

File: swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefreshIndicator.kt [hunk 1]
```diff
@@ -144,61 +102,108 @@ inline fun SwipeRefreshIndicator(
...
     shape: Shape = MaterialTheme.shapes.small.copy(CornerSize(percent = 50)),
+    refreshingOffset: Dp = 16.dp,
     largeIndication: Boolean = false,
     elevation: Dp = 4.dp,
 ) {
-    val adjustedElevation = when (offset) {
-        0f -> 0.dp
-        else -> elevation
+    val adjustedElevation = when {
+        state.isRefreshing -> elevation
```
Comment (ricknout) [inline swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefreshIndicator.kt:118]: Are these two conditions mutually exclusive?

File: swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefresh.kt [hunk 2]
```diff
@@ -232,22 +227,27 @@ fun SwipeRefresh(
     onRefresh: () -> Unit,
     modifier: Modifier = Modifier,
     swipeEnabled: Boolean = true,
-    indicator: @Composable (SwipeRefreshState) -> Unit = { SwipeRefreshIndicator(it) },
+    refreshTriggerDistance: Dp = 80.dp,
```
Comment (ricknout) [inline swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefresh.kt:230]: `80.dp` should maybe be a constant somewhere?

**Review by ricknout — APPROVED** [review]
A couple minor nits, existing demos look good and like the custom demo!

**Review by nickbutcher — APPROVED** [review]

File: swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefreshIndicator.kt (hunk 1 above)
Comment (chrisbanes) [reply swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefreshIndicator.kt:118]: Nope, but they can be exclusive

File: swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefresh.kt (hunk 2 above)
Comment (chrisbanes) [reply swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefresh.kt:230]: I'm not sure, I personally don't feel the need to outline constants for default simple parameter values. That's what control click is for. 😄

**Pull request merged**